### PR TITLE
Add test connection timeout

### PR DESCRIPTION
**Pull request by dtrifiro**
Adds an optional `timeout` kwarg to `test_connection`

### CODE REVIEW

**Comment by dtrifiro** [comment]
See https://github.com/caikit/caikit-nlp/pull/246 for more context

**Comment by dtrifiro** [comment]
Merge after #39

File: README.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
-## Caikit Text Generation Inference Service (TGIS) Backend
+# Caikit Text Generation Inference Service (TGIS) Backend
 
 This project provides a Caikit module backend that manages models run in TGIS
+
+## Configuration
+
+Sample configuration using the `MULTI` finder and a remote `TGIS` backend):
+
+```yaml
```
Comment (gabe-l-hart) [inline README.md:9]: This is great! Thanks for adding it

**Review by gabe-l-hart — APPROVED** [review]
Ship it. Very nice improvement!

**Pull request merged**